SpecFlowProject4/StepDefinitions/DynamicControlsStepDefinitions.cs

```diff
@@ -0,0 +1,48 @@
+﻿using ApiTests.Tests.UI;
+using TestProject.Tests.Pages;
+using TestProject.Utils;
+using NUnit.Framework;
+using SpecFlowProject4.StepDefinitions;
+
+namespace DynamicControlsStepDefinitions
+{
+    [Binding]
+    internal class DynamicControlsStepDefinitions : SetUp
+    {
+        readonly DynamicControlsPage dynamicControlsPage = new();
+        readonly BaseTest baseTest = new();
+        public DynamicControlsStepDefinitions(ScenarioContext _scenarioContext) : base(_scenarioContext)
+        {
+        }
+
+        [Given(@"I Click '(.*)' button")]
+        public void GivenIClickButton(string Enable)
+        {
+            baseTest.ClickOnElement(Enable);
+            dynamicControlsPage.WaitForDisableBtn();
```

Comment:
personally I'd prefer more description in the step definition if we have more than 1 action in the method. 
For example, 'I click button and wait' or in this context (since we're waiting for another element) it'd be better to separate this step into 2 smaller ones